src/main/java/com/github/cameltooling/lsp/internal/instancemodel/OptionParamURIInstance.java

```diff
@@ -102,4 +111,15 @@ public String getDescription(ComponentModel componentModel, KameletsCatalogManag
 	public CamelURIInstance getCamelUriInstance() {
 		return camelURIInstance;
 	}
+	
+	private JSONSchemaProps getKameletPropertyByKeyName(KameletsCatalogManager kameletCatalogManager, String keyName) {
+		List<Kamelet> kamelets = kameletCatalogManager.getCatalog().getKameletsByName(this.camelURIInstance.getComponentAndPathUriElementInstance().getApiNamePath().getValue());
```

Comment:
`.getApiNamePath()` should not be used in the context of a kamelet. Only in the context of an API based component. Otherwise, it is too confusing.

Either we need to come with another method with a naming matching ther Kamelet use case or use the same logic than here https://github.com/camel-tooling/camel-language-server/commit/f95cf20ed753a20f7527013be1084237544a5bf1#diff-7388766f27f5e9605ff429943a7bdd89975a182f1a6e7e94ed1e0c2cf29534dbR84

Reply:
tbh I don't understand what you are trying to achieve here.

Reply:
`getApiNamepath` was created when there was no `kamelet`. API basec components were the only one to have Camel URI properties depending on what was in the Camel URI path part apart of the component name itself. apiNamePath is misleading for kamelet context. It is currently workign but in case there is a refactor one day we need to have a distinction between an APi name path and a kamelet template it.

Reply:
I am not sure how to achieve that so I spun off a new issue for this at https://issues.redhat.com/browse/FUSETOOLS2-1156.

Reply:
see https://github.com/camel-tooling/camel-language-server/pull/611/files#r652073362 for the easy but not optimal way to do it.